tests/metrics_test.go

```diff
@@ -167,87 +158,48 @@ var _ = Describe("Metrics", func() {
 	})
 
 	Context("SSP metrics", func() {
-		var originalVersion string
-
-		waitForVersionUpdate := func(version string) {
-			Eventually(func() error {
-				rsList := &apps.ReplicaSetList{}
-				err := apiClient.List(ctx, rsList, client.InNamespace(sspDeploymentNamespace),
-					client.MatchingLabels{"name": "ssp-operator"})
-				Expect(err).ToNot(HaveOccurred())
-
-				for _, rs := range rsList.Items {
-					for _, env := range rs.Spec.Template.Spec.Containers[0].Env {
-						if env.Name == common.OperatorVersionKey &&
-							env.Value == version &&
-							*rs.Spec.Replicas > 0 &&
-							rs.Status.ReadyReplicas == *rs.Spec.Replicas {
-							return nil
-						}
-					}
-				}
-
-				return fmt.Errorf("operator version not yet updated")
-			}, env.ShortTimeout(), time.Second).Should(Succeed())
-		}
+		var templates *templatev1.TemplateList
 
-		updateVersion := func(version string) {
-			Eventually(func() error {
-				deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
-
-				envs := deployment.Spec.Template.Spec.Containers[0].Env
-
-				for i, env := range envs {
-					if env.Name == common.OperatorVersionKey {
-						envs[i].Value = version
-						break
-					}
-				}
+		getTemplates := func() *templatev1.TemplateList {
+			templates := &templatev1.TemplateList{}
+			err := apiClient.List(ctx, templates, &client.ListOptions{Namespace: getSsp().Namespace})
```

Comment:
Templates are not deployed in the same namespace as SSP CR. Please use `strategy.GetTemplatesNamespace()` to get the namespace. Also that namespace can contain templates that are not managed by SSP, so you need to check the annotations: `operator-sdk/primary-resource` and `operator-sdk/primary-resource-type`.

It would be simpler to use the code that was previously here, that used `createTestTemplate()` function.